drivers/usb/usbport/endpoint.c

```diff
@@ -1357,9 +1357,9 @@ USBPORT_InvalidateEndpointHandler(IN PDEVICE_OBJECT FdoDevice,
     {
         KeAcquireSpinLock(&FdoExtension->EndpointListSpinLock, &OldIrql);
 
-        Entry = &FdoExtension->EndpointList;
-
-        while (Entry && Entry != &FdoExtension->EndpointList)
+        for (Entry = FdoExtension->EndpointList.Flink;
+             Entry && Entry != &FdoExtension->EndpointList;
+             Entry = endpoint->EndpointLink.Flink)
```

Comment:
Can you make this `Entry = Entry->Flink` please? Doesn't feel right using `endpoint` here when that only gets initialized below.

Reply:
Simplified ;-)